src/serverenvironment.cpp

```diff
@@ -370,7 +372,7 @@ void ActiveBlockList::update(std::vector<PlayerSAO*> &active_players,
 	// Go through old list
 	for (v3s16 p : m_list) {
 		// If not on new list, it's been removed
-		if (newlist.find(p) == newlist.end())
+		if (newlist.find(p) == newlist.end() && extralist.find(p) == newlist.end())
```

Comment:
I think some of this could be simpler if the extra list was a subset of the main list.
Not 100% sure that works though

Reply:
> It's set by default and higher than active_block_range.

You're right. Sorry. 8 is fairly small, though.

> I think some of this could be simpler if the extra list was a subset of the main list.

These are the invariants:
- in the end m_list must hold all active blocks (in both ranges)
- upon exit blocks_removed must hold all removed blocks (in both ranges)
- upon exit blocks_added must hold those blocks we want emerge (those in active_block_range)
- upon exit extra_blocks_added must hold additional blocks (not found in blocks_added)

Reply:
Another way would be not to store v3s16, but a struct that includes a flag about what to do with the block (emerge or map-load only). I did not like the struct idea.

Reply:
I thought about a struct too but didn't like that either.